src/js/classes/Map.js

```diff
@@ -0,0 +1,680 @@
+    this.legend.highlightBar(bucket);
+  }
+
+  highlightMap(event) {
+    const bucket = event.srcElement.getAttribute("data-bucket");
+    this.svg.selectAll(`path:not([data-bucket="${bucket}"])`).style("opacity", "0.2");
+  }
+
+  resetHighlight() {
+    this.legend.resetHighlight();
+    this.svg.selectAll("path").style("opacity", "1");
+  }
+
+  renderPA(features, path) {
+    this.data.forEach(row => {
+      const countyName = row.data[0];
+      const cashBailRate = row.data[2];
+      const feature = features.find(f => f.properties["NAME"] === countyName);
+      feature.properties.rate = cashBailRate;
+      feature.properties.color = this.color(cashBailRate);
+      feature.properties.bucket = this.color.invertExtent(feature.properties.color)[1];
+    });
+    const paths = super.renderPA(features, path);
+    paths.style("fill", feature => feature.properties.color)
+      .attr("data-bucket", feature => feature.properties.bucket)
+      .attr("data-county-name", feature => feature.properties["NAME"])
+      .attr("data-rate", feature => feature.properties.rate);
+
+    this.legend.render();
+
+    this.renderCities();
+  }
+}
+
+class BailRaceMap extends Map {
+  constructor(selector, data, color, dataIdx, race, parent) {
+    super(selector);
+    this.data = data;
+    this.dataIdx = dataIdx;
+    this.race = race;
+    this.parent = parent;
+    this.color = color;
+  }
+
+  // Called by parent
+  _onMouseOver(countyName, x, y) {
+    const element = document.querySelector(`path[data-county-name="${countyName}"][data-race="${this.race}"]`);
+    this.svg.selectAll(`path[data-county-name="${countyName}"]`).style("stroke-width", "2px");
+    this.showTooltip(countyName, element.getAttribute("data-rate"), x, y);
+  }
+  _onMouseOut(countyName) {
+    super.onMouseOut();
+    this.svg.selectAll(`path[data-county-name="${countyName}"]`).style("stroke-width", "0.5px");
+  }
+
+  showTooltip(countyName, countyRate, pageX, pageY) {
+    super.showTooltip(pageX, pageY);
+    this.tooltip
+      .style("left", `${pageX - TOOLTIP_WIDTH}px`)
+      .style("top", `${pageY - TOOLTIP_HEIGHT}px`)
+      .html(`
+        <h3 class="tooltip-name">${countyName}</h3>
+        <table>
+          <tbody>
+            <tr>
+              <td>Cash Bail Race</td>
+              <td style="text-align: right; font-weight: 700;">${`${Math.round(countyRate * 100) / 100}%`}</td>
+            </tr>
+          </tbody>
+        </table>
+      `);
+  }
+
+  highlightMap(bucket) {
+    this.svg.selectAll(`path:not([data-bucket="${bucket}"])`).style("opacity", "0.2");
+  }
+
+  resetHighlight() {
+    this.svg.selectAll("rect").style("opacity", "1");
+    this.svg.selectAll("text").style("opacity", "1");
+    this.svg.selectAll("path").style("opacity", "1");
+  }
+
+  onMouseOver(event) {
+    this.parent.onChildMouseOver(event, this.race);
+  }
+
+  onMouseOut(event) {
+    this.parent.onChildMouseOut(event);
+  }
+
+  renderPA(features, path) {
```

Comment:
I think as long as it works, im cool with this tbh, but there's enough repetition that I think you could pull out some of this logic into separate functions.

Reply:
I can refactor this a bit, but there's enough differences between each Map implementation that it's not going to DRY very well.

Reply:
no big deal